### PR TITLE
📖 Add CABPK abbreviation and v1alpha2-ize other abbreviations

### PR DESCRIPTION
**Pull request by erwinvaneyk**
This PR adds CABPK (kubeadm bootstrap provider) to the abbreviation overview. It further rewords the abbreviations in the book to emphasize the difference between the infrastructure and bootstrap providers introduced in v1alpha2

### CODE REVIEW

File: docs/book/src/reference/abbreviations.md [hunk 1]
```diff
@@ -3,8 +3,9 @@
 Acronym or Abbreviation  | Full name
 -------------------------|---------------------
 CAPI                     | Cluster API
-CAPA                     | Cluster API Provider AWS
-CAPD                     | Cluster API Provider Docker
-CAPO                     | Cluster API Provider OpenStack
-CAPV                     | Cluster API Provider vSphere
-CAPZ                     | Cluster API Provider Azure
+CAPA                     | Cluster API AWS infrastructure provider
```
Comment (detiber) [inline docs/book/src/reference/abbreviations.md]:
```suggestion
CAPA                     | Cluster API infrastructure Provider AWS
```

File: docs/book/src/reference/abbreviations.md [hunk 2]
```diff
@@ -3,8 +3,9 @@
...
+CAPD                     | Cluster API Docker infrastructure provider
```
Comment (detiber) [inline docs/book/src/reference/abbreviations.md]:
```suggestion
CAPD                     | Cluster API infrastructure Provider Docker
```

File: docs/book/src/reference/abbreviations.md [hunk 3]
```diff
@@ -3,8 +3,9 @@
...
+CAPO                     | Cluster API OpenStack infrastructure provider
```
Comment (detiber) [inline docs/book/src/reference/abbreviations.md]:
```suggestion
CAPO                     | Cluster API infrastructure Provider OpenStack
```

File: docs/book/src/reference/abbreviations.md [hunk 4]
```diff
@@ -3,8 +3,9 @@
...
+CAPV                     | Cluster API vSphere infrastructure provider
```
Comment (detiber) [inline docs/book/src/reference/abbreviations.md]:
```suggestion
CAPV                     | Cluster API  infrastructure Provider vSphere
```

File: docs/book/src/reference/abbreviations.md [hunk 5]
```diff
@@ -3,8 +3,9 @@
...
+CAPZ                     | Cluster API Azure infrastructure provider
```
Comment (detiber) [inline docs/book/src/reference/abbreviations.md]:
```suggestion
CAPZ                     | Cluster API infrastructure Provider Azure
```

File: docs/book/src/reference/abbreviations.md [hunk 6]
```diff
@@ -3,8 +3,9 @@
...
+CABPK                    | Cluster API Kubeadm bootstrap provider
```
Comment (detiber) [inline docs/book/src/reference/abbreviations.md]:
```suggestion
CABPK                    | Cluster API Bootstrap Provider Kubeadm
```

**Comment by detiber** [comment]
doh, review in too late :(

**Comment by ncdc** [comment]
follow-up 😛

**Comment by erwinvaneyk** [comment]
@detiber it's the effort that counts! 😁

**Pull request merged**